### ISSUE TITLE
Insights from Oxygen

### ISSUE THREAD

**Issue by StreetDog1971** [pojome/elementor#4963]
I've been testing Oxygen for the past few days and there are a few features that I'd love to see in Elementor.

**1. Structure tab**
There's a plugin for Elementor (Elementor Navigation Tree) that adds a structure view and allows to select elements by name or ID. This should be in core, in more complex websites, with nested sections and columns, this is a must have.
The plugin works ok, but it won't scroll to an element when you click on its name.

**2. Carousels and sliders**
Oxygen allows you to build a carousel or a slider with whatever elements you want. Just drag stuff into a slide and that's it. Any reason why we couldn't have this in Elementor?

**3. PHP and Javascript inside elements**
Just kidding. Nice to have, for sure, but I understand the implications.

**4. CSS editor**
It's amazing the difference it makes to have a decent CSS editor inside Oxygen. Syntax highlight and a color scheme like sublime or visual studio improves CSS editing by a lot.

Keep up the great work guys! Congrats on the million active installs!

**Comment by black-eye** [comment]
Following this.
I'm in total agreement. Especially with no. 2 (I'd just add tabs and accordion/toggle to the list as well).
BTW, there's so much more to get inspired from Oxygen - e.g.: (much, much) cleaner code and classes based element styling (see: https://www.youtube.com/watch?v=oLkzurjke1k)

It would be really cool to see some of these features also in Elementor.

**Comment by maxwellhibbert** [comment]
The document Structure view in Oxygen is awesome and Elementor needs something along the lines of that Navigation Tree plugin built into the core.

But the CSS editor in Oxygen is bare bones compared with Elementor's, though - which is shocking because it's aimed at devs more so than other builders. It has no autocomplete, no multiple cursors..  it's just slower.

That being said, Oxygen has some things going for it. The main one being that all the styling is based on css and is class based, so it would be HEAPS easier to maintain or modify a design, and keep things consistent. But on the other hand, it's slower to get to the desired output..

Overall, I found Oxygen lacking when making a site faster to develop compared to doing things manually. When checking it out, I kept thinking that I might as well be putting all of this together in my code editor. There's also no Woocommerce modules, so if you want to customise any of that you'll need to hand code anyway.

Elementor could be amazing if they'd add a few more front-end developer friendly features. 

- Global styles
- Color variables
- Class based (or somehow linked) styles
- A more robust grid system
- Better/pro CSS editing in one place
- Better ACF integration

**Comment by bainternet** [comment]
@StreetDog1971 

Thanks for the feedback,
Please follow guidelines for opening a new feature request, so we can help you better.
This report is missing key information: 

* One request per GitHub issue.
* Prerequisites.

As for the requests:

1) Already implemented, waiting on testing and will be released in the next versions.

2) Duplicate of #219

3) We are not even joking about it 🤪 No real plans to implement this for So Many Reasons.

4) Duplicate of #1324 #3345

Thanks.

**Comment by StreetDog1971** [comment]
Thank you so much! I'm looking forward to the next release!!!
(Please, bring back the duplicate and delete buttons, but don't tell anyone I asked for it).
;)
